lib/oxidized/input/ssh.rb

```diff
@@ -24,20 +24,22 @@ def connect node
       secure = Oxidized.config.input.ssh.secure
       @log = File.open(Oxidized::Config::Log + "/#{@node.ip}-ssh", 'w') if Oxidized.config.input.debug?
       port = vars(:ssh_port) || 22
-      if proxy_host = vars(:ssh_proxy)
-        proxy_command =  "ssh "
-        proxy_command += "-o StrictHostKeyChecking=no " unless secure
-        proxy_command += "#{proxy_host} -W %h:%p"
-        proxy =  Net::SSH::Proxy::Command.new(proxy_command)
-      end
       ssh_opts = {
         :port => port.to_i,
         :password => @node.auth[:password], :timeout => Oxidized.config.timeout,
         :paranoid => secure,
         :auth_methods => %w(none publickey password keyboard-interactive),
         :number_of_password_prompts => 0,
-        :proxy => proxy,
       }
+
+      if proxy_host = vars(:ssh_proxy)
+        proxy_command =  "ssh "
+        proxy_command += "-o StrictHostKeyChecking=no " unless secure
+        proxy_command += "#{proxy_host} -W %h:%p"
+        proxy =  Net::SSH::Proxy::Command.new(proxy_command)
+        ssh_opts[:proxy] = proxy
+      end
+
```

Comment:
Why did you change this block from the current? Both are functional, seems unnecessary.